### PR TITLE
replicaset_mode: fix queue.tube indexes

### LINKED ISSUE #202: Failure when trying to use the tube after switching back to the master
When using the queue in the replica set, when switching from master to replica and then back again, the module does not allow to work with tubes created on the replica.

The initial settings are taken from [here](https://github.com/tarantool/queue#queue-and-replication).

master:

`box.cfg({read_only = true})`

replica:

```
box.cfg({read_only = false})
queue.create_tube('tube', 'fifo')
box.cfg({read_only = true})

```
master:

`box.cfg({read_only = false})`

and now we trying to use the `tube'.

```
tarantool> queue.tube.tube:put("a1")
---
- error: '[string "return queue.tube.tube:put("a1")"]:1: attempt to index field ''tube''
    (a nil value)'
...
```

### PR DESCRIPTION
**Pull request by 0x501D**
In replicaset mode, tubes can be created and deleted on different nodes. Accordingly, it is necessary to rebuild the queue.tube index.

Close #202

### CODE REVIEW

**Review by LeonidVas — APPROVED** [review]
Hi! Thank you for the patch.
`Close # ...` -> `Closes #`
LGTM

**Comment by 0x501D** [comment]
> Hi! Thank you for the patch. `Close # ...` -> `Closes #` LGTM

Fixed.

File: queue/abstract.lua [hunk 1]
```diff
@@ -520,10 +520,39 @@ function method._on_consumer_disconnect()
...
+        -- Accordingly, it is necessary to rebuild the queue.tube index.
+        if replicaset_mode then
+            for _, tube_name in pairs(queue.tube()) do
+                queue.tube[tube_name] = nil
+            end
+            for _, tube_tuple in box.space._queue:pairs() do
+                if queue.driver[tube_tuple[4]] ~= nil then
+                    recreate_tube(tube_tuple)
+                end
+            end
```
Comment (oleg-jukovec) [inline queue/abstract.lua:554]: You could extract it in a separate function: `recreate_tubes` - the code appears at least two times in the file.

**Review by oleg-jukovec — APPROVED** [review]
Thank you for the patch!

**Pull request merged**